src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariableSwitchStatement.java

```diff
@@ -0,0 +1,101 @@
+/*
+UnusedLocalVariable
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.coding.unusedlocalvariable;
+
+public class InputUnusedLocalVariableSwitchStatement {
+    public void testParameters0() {
+
+        int i = 0;
+        switch (i++) {
+            default:
+            case 0:
+                System.out.println("one: ");
+                break;
+            case 1:
+                System.out.println("two: " );
+                break;
+        }
+
+    }
+
+    public void testParameters01() {
+        int i = 0;
+        switch (++i) {
+            default:
+            case 0:
+                System.out.println("one: ");
+                break;
+            case 1:
+                System.out.println("two: " );
+                break;
+        }
+
+    }
+
+    public void testParameters() {
+        String[] tests = { "one", "two" };
+        int i = 0;
+        for (String test : tests) {
+            switch (i++) {
+                default:
+                case 0:
+                    System.out.println("one: " + test);
+                    break;
+                case 1:
+                    System.out.println("two: " + test);
+                    break;
+            }
+        }
+    }
+
+    public void testParameters2() {
+        String[] tests = { "one", "two" };
+        int i = 0;
+        for (String test : tests) {
+            switch (++i) {
+                default:
+                case 0:
+                    System.out.println("one: " + test);
+                    break;
+                case 1:
+                    System.out.println("two: " + test);
+                    break;
+            }
+        }
+    }
+
+    public void testParameters3() {
+        String[] tests = { "one", "two" };
+        int i = 0;
+        for (String test : tests) {
+            switch (--i) {
+                default:
+                case 0:
+                    System.out.println("one: " + test);
+                    break;
+                case 1:
+                    System.out.println("two: " + test);
+                    break;
+            }
+        }
+    }
+
+    public void testParameters4() {
+        String[] tests = { "one", "two" };
+        int i = 0;
+        for (String test : tests) {
+            switch (i--) {
```

Comment:
Add a case where switch has a different variable. Ex: var outside for is `i`, an switch is on `j`.

Can a switch have something like `i+1`?

Reply:
I have added test cases please look

Reply:
Please point out where the test cases I asked for are (including line numbers). I am not seeing them at a glance.

Reply:
The test case `i+1` -> in file InputUnusedLocalVariableSwitchStatement2.java at line 12

`Add a case where switch has a different variable. Ex: var outside for is i, an switch is on j.` as far as I understand this please check InputUnusedLocalVariableSwitchStatement2.java line 74 else I have tried to add test in https://github.com/checkstyle/checkstyle/pull/14672/files#diff-9746761dc1b35bcf722d13f1083b86a3f209e4cdd40586377c14a0299f03fa67

Reply:
Thank you, this is done.